### PR TITLE
Standardise support for reading infinite durations from Config

### PR DESCRIPTION
**Pull request by ennru**
This bit from Alpakka Kafka might be relevant across Akka.

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/Helpers.scala [hunk 1]
```diff
@@ -130,12 +130,20 @@ object Helpers {
...
+    import JavaDurationConverters._
+
     def getMillisDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.MILLISECONDS)
 
     def getNanosDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.NANOSECONDS)
 
     private def getDuration(path: String, unit: TimeUnit): FiniteDuration =
       Duration(config.getDuration(path, unit), unit)
+
+    def getPotentiallyInfiniteDuration(path: String): Duration = config.getString(path) match {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]:
```
import akka.util.Helpers.toRootLowerCase

toRootLowerCase(config.getString(path))
```

File: akka-actor/src/main/scala/akka/util/Helpers.scala [hunk 2]
```diff
@@ -130,12 +130,20 @@ object Helpers {
...
+      case "infinite" => Duration.Inf
+      case _ => config.getDuration(path).asScala
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]: should we cover `Undefined` as well, for `""` and `"off"` values?

File: akka-actor/src/main/scala/akka/util/Helpers.scala [hunk 3]
```diff
@@ -130,12 +130,20 @@ object Helpers {
    * INTERNAL API
    */
   private[akka] final implicit class ConfigOps(val config: Config) extends AnyVal {
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]: ok, this is still `private[akka]` so we must be careful to not break bin compat if used from Alpakka

Perhaps we should introduce an additional comment to "INTERNAL API" for such cases. `OptionVal` is another candidate.

File: akka-actor/src/main/scala/akka/util/Helpers.scala (hunk 2 above)
Comment (ennru) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]: Not undefined, that must be an error.

File: akka-actor/src/main/scala/akka/util/Helpers.scala (hunk 3 above)
Comment (ennru) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]: In the end of the day Alpakka should refrain from using `private[akka]` API, I guess. Just wanted to set a standard how to deal with this.

File: akka-actor/src/main/scala/akka/util/Helpers.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]: why? we use `Undefined` in some places, e.g. https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala#L89

File: akka-actor/src/main/scala/akka/util/Helpers.scala [hunk 4]
```diff
@@ -130,12 +130,20 @@ object Helpers {
...
+
     def getMillisDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.MILLISECONDS)
 
     def getNanosDuration(path: String): FiniteDuration = getDuration(path, TimeUnit.NANOSECONDS)
 
     private def getDuration(path: String, unit: TimeUnit): FiniteDuration =
       Duration(config.getDuration(path, unit), unit)
+
+    def getPotentiallyInfiniteDuration(path: String): Duration = toRootLowerCase(config.getString(path)) match {
+        case "infinite" | "" | "off" => Duration.Inf
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]: why not `Undefined` for `"" | "off"`?
we use Undefined in some places, e.g. https://github.com/akka/akka/blob/master/akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala#L89

Comment (ennru) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]: I misunderstood.

File: akka-actor/src/main/scala/akka/util/Helpers.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]: well, I actually think both cases belong to `getPotentiallyInfiniteDuration`.
Both `Duration.Undefined` and `Duration.Inf` are `Infinite` so the name is alright.

Comment (patriknw) [reply akka-actor/src/main/scala/akka/util/Helpers.scala]:
```
    def getPotentiallyInfiniteDuration(path: String): Duration = toRootLowerCase(config.getString(path)) match {
         case "infinite" => Duration.Inf
         case "" | "off" => Duration.Undefined
         case _ => config.getDuration(path).asScala
     }
```

**Comment by ennru** [comment]
Noticed that Akka HTTP's https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/impl/util/EnhancedConfig.scala contains similar extensions to Config. I guess this needs a wider approach to be useful.
Closing it for now.

**Pull request closed without merging**